### PR TITLE
Upgrade to Python 3 blog post

### PR DESCRIPTION
**Pull request by cp2004**
Here's what I've written so far in terms of a blog post, for the upgrade script. Needs a review, as well as a few bits adding in:
* Front matter images
* Details of what OctoPrint's migration took
* Date changed to when it will be published

I tried not to make it too wordy, since the more people have to read they get intimidated and won't upgrade, despite it actually being a very easy process (If I say so myself...!) 

Hopefully you should be able to push commits to my repo, to make any necessary changes.

Marked as draft PR for now, since it's not ready for release!

### CODE REVIEW

File: blog/news/_posts/2020-09-03-upgrade-to-py3.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+title: "Upgrade your OctoPrint install to Python 3!"
+author: cp2004
+# card: /assets/img/blog/  # TODO image?
+# featuredimage: /assets/img/blog/  # TODO image
+# poster: /assets/img/blog/  # TODO image
+date: 2020-09-03 12:00:00 +0200  # TODO change to publishing date
+excerpt: Give OctoPrint a spin on Python 3! Run this simple install script.
+---
+
+Since OctoPrint version 1.4.0, it has supported both Python 2 and Python 3 *simultaneously*, as a result of Python 2's EOL status. However, at the time of writing, less than 2% of tracked installs are running on Python 3.
```
Comment (fieldOfView) [inline blog/news/_posts/2020-09-03-upgrade-to-py3.md:12]: The word "simultaneously" might be confusing. It seems to suggest to me you can use OctoPrint with Python 2 and Python 3 at the same time (eg run OctoPrint with Python 3 and a plugin with Python 2). This is obviously not what you mean.

Comment (cp2004) [reply blog/news/_posts/2020-09-03-upgrade-to-py3.md:12]: @fieldOfView Thank you for reading it! 100% agree, reading it again. I will try and find a different word...

File: blog/news/_posts/2020-09-03-upgrade-to-py3.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Since OctoPrint version 1.4.0, the codebase has supported installation on both Python 2 and Python 3 environments, as a result of Python 2's EOL status. However, at the time of writing, less than 2% of tracked installs are running on Python 3.
+
+The main reason for this, was the plugin repository - there was no point installing on Python 3, if you then couldn't install any plugins! These Python 3 installs tended to be manual ones, or developers who needed to test against OctoPrint running on Python 3.
+
+First of all, there needs to be a massive shoutout to everyone who worked to make this possible, from those who contributed to core OctoPrint, to each and every plugin developer who heard the call to update their plugins to be both Python 2 & 3 compatible. As I'm writing this now, 67% of plugins are compatible with Python 3!
+
+## OctoPrint's migration to Python 3
+# TODO - Details about OctoPrint's migration?
+how long it took, (3 years? that was a long time coming!), some (brief) steps involved for those who are interested, anything else you can think of...?
```
Comment (fieldOfView) [inline blog/news/_posts/2020-09-03-upgrade-to-py3.md:20]: More interesting than how long it took to get compatibility with Python 3 is to explain what Python 2 EOL status means. IE: it does not mean that OctoPrint on Python 2 will stop working out of the blue, but if issues (eg security) with Python 2  are discovered, they will not be fixed, so it is better to use Python 3 which does receive fixes. Also mention dependencies (and possibly what a dependency is).

Comment (cp2004) [reply blog/news/_posts/2020-09-03-upgrade-to-py3.md:20]: Noted, thanks.  I agreed with @foosel on discord she would write a bit about the process, but maybe it could be an extra small section. The fact that Python 2 was released 2000 - 2010 so it is getting on for old now, may also be of note. I think a significant number of OctoPrint's dependencies are pinned to earlier versions to maintain Py2 compatibility, but not sure without some research what those are.

**Comment by foosel** [comment]
> Hopefully you should be able to push commits to my repo, to make any necessary changes.

Apparently not 😅 My suggestions:

``` patch
Index: blog/news/_posts/2020-09-03-upgrade-to-py3.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- blog/news/_posts/2020-09-03-upgrade-to-py3.md	(revision f8931f9ad1bb050f6236787124d3971707a77e8e)
+++ blog/news/_posts/2020-09-03-upgrade-to-py3.md	(revision 065079dfeba9d398bd9df0f9358898e16d87305d)
@@ -15,12 +15,40 @@
 
 First of all, there needs to be a massive shoutout to everyone who worked to make this possible, from those who contributed to core OctoPrint, to each and every plugin developer who heard the call to update their plugins to be both Python 2 & 3 compatible. As I'm writing this now, 67% of plugins are compatible with Python 3!
 
-## OctoPrint's migration to Python 3
-# TODO - Details about OctoPrint's migration?
-how long it took, (3 years? that was a long time coming!), some (brief) steps involved for those who are interested, anything else you can think of...?
+### Python 2 EOL? Python 3? What does all of this even mean?
+
+Python is the programming language that OctoPrint is primarily written in. Up until the start of the year, there were
+two versions of Python -- Python 2 and 3 -- with some big changes and incompatibilities between them. 
+Python 2 has been officially declared end-of-life (EOL) by the Python maintainers 
+as of January 1st of 2020, meaning it will no longer get any more updates (but obviously still continue to work as-is). 
 
+This meant that OctoPrint, only compatible to Python 2 up until version 1.4.0, had to become Python 3 compatible in 
+order to be future-proof (and also to profit of the new possibilities and performance gains that Python 3 offers). 
+However, as OctoPrint up until version 1.4.0 was still Python 2 exclusive and all OctoPi images up to 
+and including 0.17 also shipped with Python 2, going Python 3 exclusive was not an option either. Thus OctoPrint was 
+made compatible with both 2 **and** 3. This happened with the release of OctoPrint 1.4.0 and took well over a year and 
+the combined work of Gina and four awesome contributors to get done and shipped.
 
-## I like the sound of it! How can I upgrade?
+So as of the release of OctoPrint 1.4.0, it is compatible to both major Python versions, 2 and 3. However, the 
+underlying Python environments (e.g. the 64000 instances of OctoPi currently visible in the anonymous usage tracking)
+are still primarily Python 2.
+
+Since OctoPrint at some point will drop support for Python 2 as well sometime after March 2021 (the current 
+compatibility to both versions means an enormous amount of maintanence overhead, plus no Python 3 exclusive 
+improvements can be used), the goal is to migrate all installations out in the field -- that's you! -- to Python 3 as 
+soon as possible. Future OctoPi versions starting with 0.18 will already ship with Python 3 on board, however existing
+installations will not automatically update. They can be updated manually however, and even though not all
+plugins are yet Python 3 compatible, this might still be worth your while as Python 3 in general has better performance
+than Python 2 out of the box!
+
+### I like the sound of it! How can I upgrade?
+
 That's why I'm writing this blog post! I'm Charlie Powell ([@cp2004](https://github.com/cp2004) elsewhere), and I wrote [OctoPrint-Upgrade-To-Py3](https://github.com/cp2004/OctoPrint-Upgrade-To-Py3). It's a neat installation script that will convert your existing OctoPrint install to Python 3, hassle free. Here's how:
 
 **This script is only compatible for users running OctoPi 0.17 and above, or if you have installed OctoPrint manually you will require Python 3.6 or higher**
@@ -37,10 +65,13 @@
 Once the script is finished, reload your web interface and you should be good to go!
 
 ## Reverting to your previous install
+
 Just in case something goes wrong, I also built a script that can revert to your old install. You can find more details on the [repository here](https://github.com/cp2004/Octoprint-Upgrade-To-Py3#returning-to-the-old-install)
 
 ## Problems, questions or need help?
+
 Get in touch!
+
 * For bugs, or issues running the script, please open an issue on the [Github repository](https://github.com/cp2004/OctoPrint-Upgrade-To-Py3/issues)
 * For quick, simple questions about this script **(not support)**, please comment on the post below
 * For other support, please post in the [community forum](https://community.octoprint.org) - **please include as much detail about your setup as you can**
```

You should also pull from upstream and add your avatar URL to the authors file (just mirror what's already in there), I finally added avatars for better visibility!

**Comment by cp2004** [comment]
I'm sure I ticked the 'allow edits box'... Maybe its because it is draft? idk. Looks good, will commit those changes and mark the PR as ready-to-merge.

One last TODO remains though - the images for the thumbnail etc, I don't have the graphics necessary so *if you wouldn't mind* could you create them too?

**Comment by cp2004** [comment]
So it is done, with the exception of the image that is supposed to be in the feed on the right hand side:
image

Going to mark this as 'ready for review' to see if that means you (@foosel) will be able to push to it. Maybe? Who knows...

**Comment by cp2004** [comment]
Oh, and I need to change the publishing date, so, when is it going to be published 😁

**Comment by foosel** [comment]
I'll take care of the images tomorrow and also publish it then if that's ok with you. Beware, you might get overwhelmed with support ;)

**Comment by cp2004** [comment]
With it being 'out in the open' for a while now, I think we should be alright in terms of stability (famous last words...) - I can't do much in the morning but will be around for some of the afternoon. I think the regulars on the forum know how it works so let's cross our fingers and hope it goes well :)
